### PR TITLE
Migrate ExecutableModule to ES6 class

### PR DESCRIPTION
**Pull request by emaxx-google**
Migrate the //common/js/src/executable-module/executable-module.js file and child classes that depend on it from legacy Closure Compiler classes to ES6 classes.

This is a pure refactoring change. It's part of the effort to modernize the code base, as recent Closure Compiler versions stopped supporting the legacy class syntax.

### CODE REVIEW

File: common/js/src/executable-module/emscripten-module.js [hunk 1]
```diff
@@ -179,13 +177,13 @@ EmscriptenModule.prototype.getEmscriptenApiModuleSettings_ = function() {
...
 
 /**
  * Handles the Emscripten module's crash.
  * @private
  */
-EmscriptenModule.prototype.onModuleCrashed_ = function() {
+onModuleCrashed_() {
   goog.log.error(this.logger_, 'The Emscripten module crashed');
   this.dispose();
 };
```
Comment (Fabian-Sommer) [inline common/js/src/executable-module/emscripten-module.js:189]: nit: I think this should be `}`

File: common/js/src/executable-module/emscripten-module.js [hunk 2]
```diff
@@ -195,7 +193,7 @@ EmscriptenModule.prototype.onModuleCrashed_ = function() {
  * @param {string} message
  * @private
  */
-EmscriptenModule.prototype.onModuleStdoutReceived_ = function(message) {
+onModuleStdoutReceived_(message) {
   // TODO(#185): Consider downgrading to the "fine" level once structured log
   // messages are received from the module.
   goog.log.info(this.fromModuleMessagesLogger_, message);
```
Comment (Fabian-Sommer) [inline common/js/src/executable-module/emscripten-module.js:199]: Ditto (one line below, somehow I can't comment on the line itself)

File: common/js/src/executable-module/emscripten-module.js [hunk 3]
```diff
@@ -285,11 +279,12 @@ EmscriptenModuleMessageChannel.prototype.onMessageFromModule = function(
  * @param {!Object} message
```
Comment (Fabian-Sommer) [inline common/js/src/executable-module/emscripten-module.js:279]: Ditto (3 lines above)

File: common/js/src/executable-module/emscripten-module.js [hunk 4]
```diff
@@ -285,11 +279,12 @@ EmscriptenModuleMessageChannel.prototype.onMessageFromModule = function(
...
-EmscriptenModuleMessageChannel.prototype.sendNow_ = function(message) {
+sendNow_(message) {
   // Note: The method name must match the string in the GoogleSmartCardModule
   // Embind class definition in the entry_point_emscripten.cc files.
   // The method is accessed using the square bracket notation, to make sure that
   // Closure Compiler doesn't rename this method call.
   this.googleSmartCardModule_['postMessage'](message);
-};
+}
+}
```
Comment (Fabian-Sommer) [inline common/js/src/executable-module/emscripten-module.js:289]: This looked very confusing to me (I initially suspected a typo) - maybe a comment behind the second one that this closes EmscriptenModuleMessageChannel, or a newline might help.

File: common/js/src/nacl-module/nacl-module.js [hunk 5]
```diff
@@ -161,101 +157,105 @@ NaclModule.prototype.disposeInternal = function() {
...
-NaclModule.prototype.forceElementLoading_ = function() {
+forceElementLoading_() {
   // Request the offsetTop property to force a relayout. As of June 29, 2014,
   // this is needed if the module is being loaded in a background page (see
   // crbug.com/350445).
   // Assign the result to a random property, so that Closure Compiler doesn't
   // optimize the "useless" expression away.
   this.element_.style.top = this.element_.offsetTop;
+}
 };
```
Comment (Fabian-Sommer) [inline common/js/src/nacl-module/nacl-module.js:257]: Here too a comment on which class is being closed might be helpful

**Review by Fabian-Sommer — APPROVED** [review]
LGTM

File: common/js/src/nacl-module/nacl-module.js (hunk 5 above)
Comment (emaxx-google) [reply common/js/src/nacl-module/nacl-module.js:257]: Ack - see above.

File: common/js/src/executable-module/emscripten-module.js (hunk 1 above)
Comment (emaxx-google) [reply common/js/src/executable-module/emscripten-module.js:189]: Done here and below. Thanks for spotting.

**Pull request merged**